Userland/Libraries/LibWeb/FileAPI/Blob.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2022, Kenneth Myhra <kennethmyhra@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/StdLibExtras.h>
+#include <LibJS/Runtime/ArrayBuffer.h>
+#include <LibWeb/Bindings/DOMExceptionWrapper.h>
+#include <LibWeb/Bindings/IDLAbstractOperations.h>
+#include <LibWeb/FileAPI/Blob.h>
+
+namespace Web::FileAPI {
+
+Blob::Blob(ByteBuffer const& byte_buffer, String const& type)
+    : m_byte_buffer(move(byte_buffer))
+    , m_type(move(type))
+{
+}
+
+// https://w3c.github.io/FileAPI/#ref-for-dom-blob-blob
+NonnullRefPtr<Blob> Blob::create(Optional<Vector<BlobPart>> const& blob_parts, Optional<BlobPropertyBag> const& options)
+{
+    // 1. If invoked with zero parameters, return a new Blob object consisting of 0 bytes, with size set to 0, and with type set to the empty string.
+    if (!blob_parts.has_value() && !options.has_value())
+        return adopt_ref(*new Blob());
+
+    ByteBuffer byte_buffer {};
+    // 2. Let bytes be the result of processing blob parts given blobParts and options.
+    if (blob_parts.has_value()) {
+        auto byte_buffer_or_error = process_blob_parts(blob_parts.value());
+        if (byte_buffer_or_error.is_error())
+            warnln("Failed to process blob parts: {}", byte_buffer_or_error.error());
+        byte_buffer = byte_buffer_or_error.release_value();
+    }
+
+    String type {};
```

Comment:
Pls no introducing more null strings :catdogalert:

Reply:
Hopefully correct now
```
String type;
```

Reply:
No, that's identical. You have to say `String foo = ""` or `auto foo = String::empty()` to get an empty string instead of a null one - Kling's fault, not yours, but we want to get rid of it eventually.

Reply:
Yeah, now I remember some discussion around that issue the last few days.